aiida/engine/processes/control.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+"""Functions to control and interact with running processes."""
+from __future__ import annotations
+
+import concurrent
+import typing as t
+
+import kiwipy
+from kiwipy import communications
+from plumpy.futures import unwrap_kiwi_future
+
+from aiida.common.exceptions import AiidaException
+from aiida.common.log import AIIDA_LOGGER
+from aiida.manage.external.rmq import CommunicationTimeout
+from aiida.manage.manager import get_manager
+from aiida.orm import ProcessNode, QueryBuilder
+from aiida.tools.query.calculation import CalculationQueryBuilder
+
+LOGGER = AIIDA_LOGGER.getChild('process_control')
+
+
+class ProcessTimeoutException(AiidaException):
+    """Raised when action to communicate with a process times out."""
+
+
+def get_active_processes() -> list[ProcessNode]:
+    """Return all active processes, i.e., those with a process state of created, waiting or running.
+
+    :return: A list of process nodes of active processes.
+    """
+    filters = CalculationQueryBuilder().get_filters(process_state=('created', 'waiting', 'running'), paused=True)
+    builder = QueryBuilder().append(ProcessNode, filters=filters)
+    return builder.all(flat=True)
+
+
+def play_processes(
+    processes: list[ProcessNode] | None = None,
+    all_entries: bool = False,
+    timeout: float = 5.0,
+    wait: bool = False
+) -> None:
+    """Play (unpause) paused processes.
+
+    .. note:: Requires the daemon to be running, or processes will be unresponsive.
+
+    :param processes: List of processes to play.
+    :param all_entries: Play all paused processes.
+    :param timeout: Raise a ``ProcessTimeoutException`` if the process does not respond within this amount of seconds.
+    :param wait: Set to ``True`` to wait for process response, for ``False`` the action is fire-and-forget.
+    :raises ``ProcessTimeoutException``: If the processes do not respond within the timeout.
+    """
+    if processes and all_entries:
+        raise ValueError('cannot specify processes when `all_entries = True`.')
+
+    if not processes and all_entries:
+        processes = get_active_processes()
+
+    if not processes:
+        raise ValueError('no processes specified and `all_entries = False`.')
```

Comment:
I don't think that's true, as `get_active_processes()` may also just have returned an empty list above.

Maybe easier to just say "no processes selected".

And I would consider this a valid selection, i.e. I would not raise an error but rather use something like `LOGGER.report` (or warning) and `return`